tsl/test/expected/dist_ddl.out

```diff
@@ -475,9 +487,19 @@ ERROR:  cannot drop a hypertable along with other objects
 DROP TABLE disttable, disttable;
 ERROR:  cannot drop a hypertable along with other objects
 \set ON_ERROR_STOP 1
---------------------------------------------------------------------
--- Test renaming columns, constraints, indexes, and REINDEX command.
---------------------------------------------------------------------
+----------------------------------------------------------------------------------------
+-- Test column type change, renaming columns, constraints, indexes, and REINDEX command.
+----------------------------------------------------------------------------------------
+ALTER TABLE disttable ALTER COLUMN description TYPE VARCHAR(10);
+ALTER TABLE disttable ADD COLUMN float_col float;
+ALTER TABLE disttable ALTER COLUMN float_col TYPE INT USING float_col::int;
+\set ON_ERROR_STOP 0
+-- Changing the type of a hash-partitioned column should not be supported
+ALTER TABLE disttable ALTER COLUMN temp TYPE numeric;
+ERROR:  cannot change the type of a hash-partitioned column
+\set ON_ERROR_STOP 1
+-- Should be able to change if not hash partitioned though
+ALTER TABLE disttable ALTER COLUMN time TYPE timestamp;
```

Comment:
Are we sure it is safe to do this? For instance, changing the time type from bigint to int might change the min/max allowed value and expand or shrink the check constraints on chunks.

I think it might be safest to block changes in types of all partitioning columns. Perhaps going from int to bigint is safe, but we'd have to think through the cases.

I am actually surprised this isn't already blocked in TimescaleDB. Doesn't seem to be a multi-node specific issue.

Reply:
I was actually following the same test used in `test/partition.sql` for this

Reply:
It is not safe to do in general, it will only work for a limited subset of alter table alter type